src/main.rs

```diff
@@ -205,25 +205,287 @@ fn main() {
+}
+	
+                 if level.map == 1 {
+                 level.behavior = match level.map {
+	             1 => if rng.gen_ratio(1, 4) { 0 } else if rng.gen_ratio(1, 2) { 1 } else { 4 }
+				 4 => if rng.gen_ratio(1, 2) { 0 } else { 1 } //Elevator Behavior Softlock Fix
+				 3 => if rng.gen_ratio(1, 2) { 0 } else { 1 } //Tennis Stands Behavior Softlock Fix
+	             _ => rng.sample(Uniform::new_inclusive(4, 5)),
+                };
+	}
                 level.zoom_out_max = rng.sample(Uniform::new_inclusive(-406.0, -135.0));
                 level.zoom_in_max = rng.sample(Uniform::new_inclusive(-135.0, -22.0));
 
                 level.darkness = if rng.gen_ratio(1, 2) {
                     0.0 // 50% chance for no darkness
                 } else {
-                    rng.sample(Uniform::new_inclusive(38.0, 90.0))
+                    rng.sample(Uniform::new_inclusive(84.0, 110.0))
                 };
                 level.head_size = rng.sample(Uniform::new_inclusive(1.35, 3.5));
-            }
+				if level.darkness > 1.0
+				{level.unk12 = 3.0;}
+				{level.unk13 = 10.0;}
+				{level.unk14 = 1.0;}
+				{level.unk15 = 0.0;}
+				{level.unk16 = 6.0;}
+				
+				if level.map == 3 { 
+				level.zoom_out_max = if rng.gen_ratio(1, 2) { -279.60767 } else { -187.60767 };
+				level.zoom_in_max = -279.60767;
+				}
+				
+				if level.map == 3 { 
+				level.horiz_dist = 4.7;
+				level.vert_dist = 22.4;
+				}
+				
+				if level.map == 4 { 
+				level.unk7 = 35.0;
+				
+				}
+				
+			    if level.map == 2 {
+				level.unk7 = 0.0;
+				
+				
+				}
+				if level.map == 3 {
+				level.unk7 = -60.0;
+				
+				}
+				//Street Mii Bounding Fixes
+			    if level.map == 0 { 
+                 level.zoom_out_max = if rng.gen_ratio(3, 4) { -199.0 } else { -205.0 };
+				 level.zoom_in_max = if rng.gen_ratio(3, 4) { -199.0 } else { -205.0 };
+			     level.horiz_dist = 35.0;
+				  level.vert_dist = 97.0;
+				}
+				//Ocean Mii Bounding Fixes
+			    if level.map == 1 { 
+                 level.zoom_out_max = if rng.gen_ratio(3, 4) { -180.0 } else { -240.0 };
+				 level.zoom_in_max = if rng.gen_ratio(3, 4) { -180.0 } else { -240.0 };
+			     level.horiz_dist = if rng.gen_ratio(3, 4) { 30.0 } else { 40.0 };
+				  level.vert_dist = if rng.gen_ratio(3, 4) { 90.0 } else { 113.0 };
+			
+				}
+				//Better safe than sorry
+				// If you have too many miis on the Escalator the game will crash
+		        if level.map == 4 { // If Escalator Stage
+                level.num_miis = rng.sample(Uniform::new_inclusive( 8, 40)); // Nothing more then 40 miis
+				
+				}
+				if level.level_type == 5 {
+				
+				level.behavior = 1;
+				}
+				
+				if level.level_type == 9 {
+				level.behavior = 1;
+				
+				}
+				
+				if level.level_type == 10 {
+				level.behavior = 1;
+				
+				}
+				
+				if level.level_type == 11 {
+				level.behavior = 1;
+				
+				}
+				
+				if level.level_type == 20 {
+				level.behavior = 1;
+				
+				}
```

Comment:
shouldn't this already be handled by some code above? If it isn't, then it should be merged into the code above.

Reply:
Yeah, but it didn't work until I added this. Some level types only work with certain behaviors.

Reply:
it's probably because you're setting it up there, and then overwriting it by setting the level behavior again.

Should I try to make it so that we have a list of possible values that we remove from? (i.e. every loop we start with a behavior map or something, and then for level type 10 we remove all behaviors except 1, etc.)

Reply:
yeah, that might be a good idea